src/main/java/com/synopsys/integration/jenkins/JenkinsProxyHelper.java

```diff
@@ -40,34 +40,53 @@
 import jenkins.model.Jenkins;
 
 public class JenkinsProxyHelper {
-    public static ProxyInfo getProxyInfoFromJenkins(final String url) {
-        final ProxyConfiguration proxyConfig = Optional.ofNullable(Jenkins.getInstanceOrNull())
-                                                   .map(jenkins -> jenkins.proxy)
-                                                   .orElse(null);
-        if (proxyConfig == null) {
-            return ProxyInfo.NO_PROXY_INFO;
-        }
-
-        String username = null;
-        String ntlmDomain = null;
-        if (StringUtils.isNotBlank(proxyConfig.getUserName())) {
-            final String[] possiblyDomainSlashUsername = proxyConfig.getUserName().split(Pattern.quote("\\"));
+    private String url;
+    private String proxyHost;
+    private int proxyPort;
+    private String proxyUsername;
+    private String proxyPassword;
+    private List<Pattern> ignoredProxyHosts;
+    private String ntlmDomain;
+    private String ntlmWorkstation;
+    private ProxyInfo proxyInfo = ProxyInfo.NO_PROXY_INFO;
+
+    public JenkinsProxyHelper(final ProxyConfiguration proxyConfiguration, final String url) {
```

Comment:
I think this constructor should be changed to a static method to construct a JenkinsProxyHelper.  From the fields listed, the second constructor seems more appropriate and this seems more like it should be a static helper method to construct a JenkinsProxyHelper using the ProxyConfiguration.